### PR TITLE
Extending directors crud

### PR DESCRIPTION
**Pull request by Joey-2134**
think added all functionality for director, needs to be tested though

### CODE REVIEW

File: src/main/java/com/cs5106/movieMuseum/domain/controller/DirectorController.java
Comment (seanieStack) [inline src/main/java/com/cs5106/movieMuseum/domain/controller/DirectorController.java:1]: Not sure what is taught in your class but i prefer using @RequestMapping and specifing the request type in there as such

@GetMapping("/directors")

becomes 

@RequestMapping(value = "/directors/{id}", method = RequestMethod.GET)

Allthough for a project of this simplicity GetMapping is probaly better suited

Abstract business logic in a service class, controller should be concice and simple to understand inputs and outputs

File: src/main/java/com/cs5106/movieMuseum/domain/controller/ActorController.java
Comment (seanieStack) [inline src/main/java/com/cs5106/movieMuseum/domain/controller/ActorController.java:1]: Not sure what is taught in your class but i prefer using @RequestMapping and specifing the request type in there as such

@GetMapping("/directors")

becomes

@RequestMapping(value = "/directors/{id}", method = RequestMethod.GET)

Allthough for a project of this simplicity GetMapping is probaly better suited

Abstract business logic in a service class, controller should be concice and simple to understand inputs and outputs

File: src/main/java/com/cs5106/movieMuseum/MovieMuseumApplication.java
Comment (seanieStack) [inline src/main/java/com/cs5106/movieMuseum/MovieMuseumApplication.java:1]: I understand this is tempory, but no just no

**Review by seanieStack — COMMENTED** [review]
7/10

File: src/main/java/com/cs5106/movieMuseum/domain/controller/DirectorController.java
Comment (seanieStack) [reply src/main/java/com/cs5106/movieMuseum/domain/controller/DirectorController.java:1]: you can also add 

@RequestMapping("/directors") 

at the top to so all following methods can become

@GetMapping("/{id}")

but this assumes your are always going to use ./directors/ and some variation such as ./director/

Comment (seanieStack) [reply src/main/java/com/cs5106/movieMuseum/domain/controller/DirectorController.java:1]: image

Comment (seanieStack) [reply src/main/java/com/cs5106/movieMuseum/domain/controller/DirectorController.java:1]: clean easy to understand for acces in frontend / postman request testing

**Comment by jaypopat** [comment]
LGTM

**Pull request merged**